src/bpmncwpverify/visitors/bpmn_connectivity_visitor.py

```diff
@@ -0,0 +1,51 @@
+from bpmncwpverify.core.bpmn import (
+    BpmnVisitor,
+    Flow,
+    Node,
+    SequenceFlow,
+    StartEvent,
+    EndEvent,
+    IntermediateEvent,
+)
+
+
+class BpmnConnectivityVisitor(BpmnVisitor):  # type: ignore
+    def _ensure_in_messages(self, node: Node, obj_type: str) -> None:
+        if node.in_msgs:
+            if not node.message_event_definition:
+                raise Exception(
```

Comment:
Do these exception get caught and turned into specific `Error` types?

If no, then we need to create the appropriate `Error` types with their corresponding `get_message` entries so that our error handing is uniform. All errors should be communicated through the `Result` type and the message should be constructed from information in the associated wrapped `Error` type.

Reply:
So you'll need an `Error` subclass for each of these, and that sub-class should have the `obj_type` and `node.id` as attributes so the `get_error_message` can construct itself.